user-contrib/Ltac2/tac2core.ml

```diff
@@ -1679,6 +1679,16 @@ let () = add_scope "constr" (fun arg ->
     Tac2entries.ScopeRule (Pcoq.Symbol.nterm Pcoq.Constr.constr, act)
   )
 
+let () = add_scope "open_constr" (fun arg ->
+    let delimiters = List.map (function
+        | SexprRec (_, { v = Some s }, []) -> s
+        | _ -> scope_fail "open_constr" arg)
+        arg
+    in
+    let act e = Tac2quote.of_open_constr ~delimiters e in
+    Tac2entries.ScopeRule (Pcoq.Symbol.nterm Pcoq.Constr.constr, act)
+  )
+
```

Comment:
I would suggest factoring this code out into an `add_scope_with_delimiters` or something like that, and then using it for `constr`, `open_constr`, and ideally for `pattern` too.  (If you think this is a bad idea, I'm also fine merging without this.)

Reply:
It's not very easy to do without essentially parameterizing everything. The only common part is the parsing function and even there the error is not the same in both cases. So I think it's not worth the effort.